llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -1971,6 +1971,57 @@ bool AArch64TargetLowering::shouldExpandGetActiveLaneMask(EVT ResVT,
   return false;
 }
 
+bool AArch64TargetLowering::shouldExpandPartialReductionIntrinsic(
+    const CallInst *CI) const {
+  const bool TargetLowers = false;
+  const bool GenericLowers = true;
+
+  auto *I = dyn_cast<IntrinsicInst>(CI);
+  if (!I)
+    return GenericLowers;
+
+  ScalableVectorType *RetTy = dyn_cast<ScalableVectorType>(I->getType());
```

Comment:
Am I right that this is just an artificial limitation at the moment? (i.e. we can make this work for fixed-length vectors too)

Reply:
Yeah you're right. I've removed that limitation now.